File: Java/Medium/WrongAnswer/range-product-queries-of-powers_1.java
class Solution {
    public int[] productQueries(int n, int[][] queries) {
        
        int[] powers = new int[31];
        for (int i = 0; i < 31; i++) {
            powers[i] = 1 << i;
        }
        int[] ret = new int[queries.length];
        for (int i = 0; i < queries.length; i++) {
            int left = queries[i][0];
            int right = queries[i][1];
            int product = 1;
            for (int j = left; j <= right; j++) {
                product *= powers[j];
                if (product > 1000000007) {
                    product %= 1000000007;
                }
            }
            ret[i] = product;
        }
        return ret;
    }
}

File: Rust/Medium/WrongAnswer/range-product-queries-of-powers_1.rs
impl Solution {
    pub fn product_queries(n: i32, queries: Vec<Vec<i32>>) -> Vec<i32> {
        
        let mut powers = Vec::new();
        let mut n = n;
        let mut i = 0;
        while n > 0 {
            if n & 1 == 1 {
                powers.push(i);
            }
            n >>= 1;
            i += 1;
        }
        let mut res = Vec::new();
        for query in queries {
            let mut prod = 1;
            for i in powers.iter().skip(query[0] as usize).take((query[1] - query[0] + 1) as usize) {
                prod *= 1 << i;
                prod %= 1_000_000_007;
            }
            res.push(prod);
        }
        res
    }
}